cpp/src/io/utilities/column_buffer.cpp

```diff
@@ -78,7 +79,19 @@ std::unique_ptr<column> make_column(column_buffer& buffer,
         // convert to binary
         auto const string_col = make_strings_column(*buffer._strings, stream, mr);
         auto const num_rows   = string_col->size();
-        auto col_contest      = string_col->release();
+        auto col_content      = string_col->release();
+
+        // convert to uint8 column, strings are currently stores as int8
+        auto contents =
+          col_content.children[strings_column_view::chars_column_index].release()->release();
+        auto data      = contents.data.release();
+        auto null_mask = contents.null_mask.release();
+
+        auto uint8_col = std::make_unique<column>(data_type{type_id::UINT8},
+                                                  data->size(),
+                                                  std::move(*data),
+                                                  std::move(*null_mask),
+                                                  UNKNOWN_NULL_COUNT);
```

Comment:
It seems like this is the bit that's confounding `spark-rapids`. In doing this, the metadata section in the output goes from:
```
FileMetaData{schema: message schema {
  required binary mybin;
}
```
... to this:
```
FileMetaData{schema: message schema {
  required group mybin (LIST) {
    repeated group list {
      required int32 element (INTEGER(8,true));
    }
  }
}
```
The relabeling is breaking Spark's data schema inference logic.

Reply:
The way I tested this before was to write out `STRING` data to Parquet as `BINARY`, and then attempt to read it back. The corresponding Spark code was:
```scala
val filepath = "/tmp/strbin.parquet"
sql(" SELECT CAST('Hello, World' AS BINARY) AS strbin ").write.parquet(filepath)
spark.read.parquet(filepath).show
```

Have we confirmed that this works, after this change?

Reply:
> This was (surpisingly) not the issue causing the schema inconsistencies -- that was an implicit call on the spark side to a cast to list of binary that was not doing this kind of uint8 conversion. 

In my example above, if the user casts explicitly to `BINARY` before writing to Parquet, then they expect to read `BINARY` when the output file is read. If it reads back as `LIST<BYTE>`, I don't think the change is right.

Reply:
>  If it reads back as `LIST<BYTE>`, I don't think the change is right.

We just tested this. The column is written correctly as `BINARY`. This should appease Spark.